src/Routes/RosPage/RosPage.js

```diff
@@ -124,7 +130,46 @@ class RosPage extends React.Component {
         ).then((results) => results);
     }
 
+    updateStateFilter = (value) => {
+        this.setState({
+            stateFilterValue: value
+        });
+    }
+
+    onDeleteFilters = (e, filtersArr) => {
+        const deletedStateFilters = filtersArr.filter((filterObject) => {
+            return filterObject.category === 'State';
+        });
+
+        if (deletedStateFilters.length > 0) {
+            const resetFiltersList = deletedStateFilters[0]?.chips.map((chip) =>{
+                return chip?.name;
+            });
+
+            const activeStateFilters = this.state.stateFilterValue.filter(filterName => !resetFiltersList.includes(filterName));
+
+            this.setState ({
+                stateFilterValue: activeStateFilters
+            });
+        }
+    }
+
+    getActiveFilterConfig = () => {
+        let activeFilters = this.state.stateFilterValue.map((value)=> {
+            return { name: value };
+        });
+
+        return activeFilters.length > 0
+            ? [{
+                category: 'State',
+                chips: activeFilters
+            }]
+            : [];
+    }
+
     renderConfigStepsOrTable() {
+        const { state: SFO } = CUSTOM_FILTERS;
```

Comment:
SFO? 
I understood it's just a variable name but will it be good idea to rename it to something meaningful?

Reply:
I agree @kgaikwad renamed it to SFObject 💯